unit_tests/test_cluster.py

```diff
@@ -974,6 +974,25 @@ def is_virtual_tables_get_snapshot():
                                  f"Actual snapshot content: {sorted(snapshot_content_list)}")
 
 
+@pytest.mark.integration
+def test_exclusive_connection(docker_scylla, docker_scylla_2, params, events):
+    """
+    Test exclusive CQL connection creation for each node in the cluster.
+    Ensures that the session connects to the correct node.
+    Run 10 times to increase the chance of catching intermittent issues.
+    """
+    cluster = DummyScyllaCluster([docker_scylla, docker_scylla_2])
```

Comment:
Those are two separate scylla instances, they are not configured as a cluster, so it's a bit lacking in that sense.

Reply:
I ran this test using cql_connection instead of cql_connection_exclusive, and the test failed because the selected node was random and didn't match the one set for cql_connection. So, this behaviour appears to be expected.

Reply:
changed